File: array/724.py
class Solution:
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        total = sum(nums)
        parta = 0
        for i in range(len(nums)):
            if (total - nums[i]) == parta:
                return i
            parta += nums[i]
            total -= nums[i]
        return -1   
            

File: array/724.java
class Solution {
    public int pivotIndex(int[] nums) {
        if (nums.length <= 2) {
            return -1;
        }
        int leftsum = 0;
        int rightsum = 0;
        
        for (int i = 1; i < nums.length; i++) {
            rightsum += nums[i];
        }
        for (int i = 0; i < nums.length; i++) {
            if (leftsum == rightsum) {
                return i;
            }else {
                leftsum += nums[i];
                if (i < nums.length-1) {
                    rightsum -= nums[i+1];    
                }
            }
        }
        return -1;
    }
}
